src/app/problem.cljs

```diff
@@ -67,17 +67,22 @@
     (let [next-prob (next-problem id)
           on-run (fn []
                    (try
-                       (let [attempts (check-solution problem (get-editor-value))]
-                         (when attempts
-                           (reset! results attempts)
-                           (reset! modal-is-open true)))
-                       (catch ExceptionInfo e
-                         (reset! error-stacktrace (-> e ex-data :stacktrace)))))]
+                     (let [editor-value (get-editor-value)
+                           _ (reset! code editor-value)
+                           attempts (check-solution problem editor-value)]
+                       (when attempts
+                         (reset! results attempts)
+                         (reset! modal-is-open true)))
+                     (catch ExceptionInfo e
+                       (reset! error-stacktrace (-> e ex-data :stacktrace)))))]
       [:div
        (when (:restricted problem)
          [restricted-alert problem])
        [:p "Write code which will fill in the above blanks:"]
-       [editor/editor @code !editor-view {:eval? true}]
+
+       ;; Force resetting editor state when input source code changed
+       ;; e.g., when manually trigger run
+       ^{:key @code} [editor/editor @code !editor-view {:eval? true}]
```

Comment:
I have a question about this. My first attempt is to re-calculate `(sci/eval-string source)` in https://github.com/oxalorg/4ever-clojure/blob/main/src/app/editor.cljs#L59, but I am not sure if this is possible with Reagent/with-let that according to the API this expression will only be called once in the component's lifespan.
I assume there is a better way to re-eval with-let based on some dependencies though ... However I am new to Clojure & Reagent.

Reply:
Yes that is why `last-result` is an atom, so that we can change it. And yes you are correct anything in `with-let` can't be modified (unless it's an atom). 

To do that we need to use `form-3` components and use the lifecycle method `did-component-update`. I'm not a huge fan of those though.

Reply:
Thanks for the directions! at first I thought reaction/track sounds the cue which both look like something equivalent to useEffect in React, but this was not true. I then found out the form-3 looks like the real solution to interpolate with React lifecycles, but yes it looks ugly! So in the end I chose to use key to force recreating the editor instead, 🤔

Btw I saw somewhere that 4ever clojure is rewriting with re-frame. I also have very limited knowledge on Rum. They are all quite similar to me, but could you briefly tell me the reason why you chose reframe at the end?

Thanks!

Reply:
Exploring CLJS web apps building with React under the hood is really inspiring and eye-opening. I will take some more time to learn the different counterparts in Clojure ecosystem.
Again, thanks for your guidance!

Reply:
Thank you for being awesome :) Feel free to join the clojurians slack community: http://clojurians.net/, lots of discussion happens there! I recently tried to tag you in there as a contributor to 4ever-clojure too! :)

Reply:
Thanks! I have joined the slack channel :D There is really a ton of good things to learn in this community!